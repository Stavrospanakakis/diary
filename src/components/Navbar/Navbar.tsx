import React from 'react'
import NavButton from './navButton'

const Navbar: React.FC = () => {
    return (
        <nav className='navbar'>
            <ul className='navbar__ul'>
                <NavButton 
                    href='/'
                    name='Home'
                />
            </ul>
        </nav>
    )
}

export default Navbar