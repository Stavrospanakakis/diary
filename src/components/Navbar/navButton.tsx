import React from 'react'

interface INavButtonProps {
    href: string
    name: string
}
const NavButton: React.FC<INavButtonProps> = props => {
	const { href, name } = props
	return (
		<a href={href}>{name}</a>
	)
}

export default NavButton