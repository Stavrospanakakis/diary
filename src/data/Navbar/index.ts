const NavbarData = {
    title: 'Diary of a Web Developer',
    pages: [
        {
            name: 'Projects',
            href: 'https://github.com'
        },
        {
            name: 'Resume',
            href: 'https://resume.com'
        },
    ]
}

export default NavbarData