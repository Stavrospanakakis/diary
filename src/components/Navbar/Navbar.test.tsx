import React from 'react'

import { render } from '@testing-library/react'

import Navbar from './Navbar'

describe('Navbar', () => {
	it('should render', () => {
		const { container } = render(<Navbar />)
		expect(container).toMatchSnapshot()
	})
})