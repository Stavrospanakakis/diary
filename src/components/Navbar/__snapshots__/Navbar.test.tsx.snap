// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div>
  <div
    class="bg-white shadow-lg"
  >
    <div
      class="xl:container xl:mx-auto sm:flex sm:items-center sm:px-4 sm:py-3 flex items-center justify-center md:justify-between px-4 py-3 sm:p-0"
    >
      <div
        class="text-center"
      >
        <a
          class="text-purple-700 font-bold uppercase"
          href="/"
        >
          Diary of a Web Developer
        </a>
      </div>
      <div
        class="px-2 pt-2 pb-4 sm:flex sm:p-0"
      >
        <a
          class="mt-1 block px-2 py-1 text-purple-700 rounded hover:text-purple-900 sm:mt-0 sm:ml-2 hidden md:flex"
          href="https://github.com/Stavrospanakakis"
        >
          Projects
        </a>
        <a
          class="mt-1 block px-2 py-1 text-purple-700 rounded hover:text-purple-900 sm:mt-0 sm:ml-2 hidden md:flex"
          href="https://stavrospanakakis.com/cv/"
        >
          Resume
        </a>
      </div>
    </div>
  </div>
</div>
`;
